### PR TITLE
Bk cnn

### PR DESCRIPTION
**Pull request by evanyeohboonkhai**

### CODE REVIEW

**Comment by evanyeohboonkhai** [comment]
Please have a review on 

- 04-CNN-Univariate-Single-Step.ipynb
- 04-CNN-Univariate-Multi-Step.ipynb
- 04-CNN-Multivariate-Univariate-Single-Step.ipynb
- 04-CNN-Multivariate-Univariate-Multi-Step.ipynb
- 04 - Hybrid-CNN-LSTM Univariate.ipynb
@scot-certifai @KianYang-Lee

**Comment by KianYang-Lee** [comment]
Got conflict, please resolve it first thanks @evanyeohboonkhai

File: environment.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
-name: time-series-labs
```
Comment (KianYang-Lee) [inline environment.yml:1]: This environment.yml is outdated. Please update.

File: notebook/Solution/04 - CNN-Univariate-Single-Step.ipynb [hunk 2]
```diff
@@ -0,0 +1,3828 @@
+{
```
Comment (KianYang-Lee) [inline notebook/Solution/04 - CNN-Univariate-Single-Step.ipynb:1]: Looks good.

File: notebook/Solution/data_module.py [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+import numpy as np
+import torch
+import matplotlib.pyplot as plt
+import math
+import pandas as pd
+from torch.utils.data import DataLoader,TensorDataset
+from sklearn.metrics import mean_squared_error
+from matplotlib.lines import Line2D
+
+# In[1]:
```
Comment (KianYang-Lee) [inline notebook/Solution/data_module.py:16]: please Remove redundant lines

File: notebook/Solution/deep_learning_module.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[2]:
```
Comment (KianYang-Lee) [inline notebook/Solution/deep_learning_module.py:4]: pls remove redundant lines

File: notebook/Solution/04 - CNN-Univariate-Multi-Step.ipynb [hunk 5]
```diff
@@ -0,0 +1,3399 @@
+{
```
Comment (KianYang-Lee) [inline notebook/Solution/04 - CNN-Univariate-Multi-Step.ipynb:1]: Looks good to merge for me.

File: notebook/Solution/04 - CNN-Multivariate-Univariate-Multi-Step.ipynb [hunk 6]
```diff
@@ -0,0 +1,2926 @@
+{
```
Comment (KianYang-Lee) [inline notebook/Solution/04 - CNN-Multivariate-Univariate-Multi-Step.ipynb:1]: Looks good to merge for me.

File: notebook/Solution/04 - CNN-Multivariate-Univariate-Single-Step.ipynb [hunk 7]
```diff
@@ -0,0 +1,3029 @@
+{
```
Comment (KianYang-Lee) [inline notebook/Solution/04 - CNN-Multivariate-Univariate-Single-Step.ipynb:1]: Looks good to merge for me.

File: notebook/Solution/04 - Hybrid-CNN-LSTM Univariate.ipynb [hunk 8]
```diff
@@ -0,0 +1,3226 @@
+{
```
Comment (KianYang-Lee) [inline notebook/Solution/04 - Hybrid-CNN-LSTM Univariate.ipynb]: Looks good to merge imo.

**Review by KianYang-Lee — APPROVED** [review]
pls add files into Training folder too thanks.

**Review by scot-certifai — APPROVED** [review]
Second to all that KY had already commented. One additional comment would be on the file naming convention. If there is already a space between the number and the dash, then no point for us to use (-) as delimiters. 

Maybe we can just use space to keep it clean? Since our labs notebook will rarely be called from the terminal, I see no harm in using space. (although its not the best practice)

**Comment by evanyeohboonkhai** [comment]
> Second to all that KY had already commented. One additional comment would be on the file naming convention. If there is already a space between the number and the dash, then no point for us to use (-) as delimiters.
> 
> Maybe we can just use space to keep it clean? Since our lab's notebook will rarely be called from the terminal, I see no harm in using space. (although it's not the best practice)

I think we keep as the original naming convention to prevent any broken link within the notebook [some of my notebooks have  linked to others notebook, I am not sure about KY]

**Comment by KianYang-Lee** [comment]
I have no issue with renaming. Up to you 2 to decide. Thanks

**Comment by evanyeohboonkhai** [comment]
@KianYang-Lee  Please merge it if ok for you

**Review by scot-certifai — APPROVED** [review]
Good to merge.

**Pull request merged**